packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,157 @@
+import fs from 'fs';
+import path from 'path';
+import { AxiosError } from 'axios';
+import * as crypto from 'node:crypto';
+import { apiConfig } from '../config/api';
+import { compressFilesToTar } from '../utils/compress-files';
+import createProjectAction from '../create-project/action';
+import type { CLIContext, ProjectInfos } from '../types';
+import { getTmpStoragePath } from '../config/local';
+import { cloudApiFactory, tokenServiceFactory, local } from '../services';
+import { notificationServiceFactory } from '../services/notification';
+import { loadPkg } from '../utils/pkg';
+import { buildLogsServiceFactory } from '../services/build-logs';
+
+const FILE_SIZE_LIMIT = 100 * 1024 * 1024; // 100MB
+
+type PackageJson = {
+  name: string;
+  strapi?: {
+    uuid: string;
+  };
+};
+
+async function upload(ctx: CLIContext, project: ProjectInfos, token: string) {
+  const cloudApi = cloudApiFactory(token);
+  // * Upload project
+  try {
+    const storagePath = getTmpStoragePath();
+    const projectFolder = path.resolve(process.cwd());
+    const packageJson = (await loadPkg(ctx)) as PackageJson;
+
+    if (!packageJson) {
+      ctx.logger.error(
+        'Unable to deploy the project. Please make sure the package.json file is correctly formatted.'
+      );
+      return;
+    }
+
+    ctx.logger.log('📦 Compressing project...');
+    // hash packageJson.name to avoid conflicts
+    const hashname = crypto.createHash('sha512').update(packageJson.name).digest('hex');
+    const compressedFilename = `${hashname}.tar.gz`;
+    try {
+      ctx.logger.debug(
+        'Compression parameters\n',
+        `Storage path: ${storagePath}\n`,
+        `Project folder: ${projectFolder}\n`,
+        `Compressed filename: ${compressedFilename}`
+      );
+      await compressFilesToTar(storagePath, projectFolder, compressedFilename);
+      ctx.logger.log('📦 Project compressed successfully!');
```

Comment:
Any reason not to use the stream API to compress, tar & send for deployment directly ?

Reply:
> Any reason not to use the stream API to compress, tar & send for deployment directly ?

No particular reason (from what I know, we just went for the first solution that came to mind).

From what I've tried though, switching to “stream only” might add too much complexity, mainly because it looks challenging to get the file size beforehand:
	- we would either need to go through the compress transformer twice to know the size beforehand (which is not optimal since it means compressing twice)
	- or we could get back to the user with the “file too big” error only after the upload hits the limit (which can take some time => not good UX)
 - maybe there is a different solution that I'm not aware of?

We can still keep the idea for later improvement if we have a proper solution for the file size & once we have more time to change the logic 👍 for now I believe the best is to go ahead with the current implementation (considering time constraint + benefit of this change would be minimal imo)